ISL.Providers.Storages.AzureBlobStorage/Services/Foundations/Storages/StorageService.cs

```diff
@@ -199,13 +201,36 @@ public ValueTask<string> CreateDirectorySasTokenAsync(
             return sasToken;
         });
 
-        virtual internal string ConvertPolicyNameToPermissions(string policyName) => policyName switch
+        virtual internal string ConvertToPermissionsString(List<string> permissions)
         {
-            "read" => "rl",
-            "write" => "w",
-            "delete" => "d",
-            "fullaccess" => "rlwd",
-            _ => ""
-        };
+            string permissionsString = "";
+
+            if (permissions.Contains("read", StringComparer.OrdinalIgnoreCase))
+            {
+                permissionsString = permissionsString + "r";
+            }
+            if (permissions.Contains("add", StringComparer.OrdinalIgnoreCase))
+            {
+                permissionsString = permissionsString + "a";
+            }
+            if (permissions.Contains("create", StringComparer.OrdinalIgnoreCase))
+            {
+                permissionsString = permissionsString + "c";
+            }
+            if (permissions.Contains("write", StringComparer.OrdinalIgnoreCase))
+            {
+                permissionsString = permissionsString + "w";
+            }
+            if (permissions.Contains("delete", StringComparer.OrdinalIgnoreCase))
+            {
+                permissionsString = permissionsString + "d";
+            }
+            if (permissions.Contains("list", StringComparer.OrdinalIgnoreCase))
+            {
+                permissionsString = permissionsString + "l";
+            }
+
+            return permissionsString;
+        }
```

Comment:
Too many if's.  This can be simplified like this:


```suggestion
        virtual internal string ConvertToPermissionsString(List<string> permissions)
        {
            var permissionMap = new Dictionary<string, char>(StringComparer.OrdinalIgnoreCase)
            {
                { "read", 'r' },
                { "add", 'a' },
                { "create", 'c' },
                { "write", 'w' },
                { "delete", 'd' },
                { "list", 'l' }
            };

            return new string(permissions
                .Where(permission => permissionMap.ContainsKey(permission))
                .Select(permission => permissionMap[permission])
                .ToArray());
        }
```

Reply:
This has been refactored